### PR TITLE
Smil support added

### PR DESCRIPTION
**Pull request by paulinthought**

### CODE REVIEW

File: src/sdk/helpers/rect.js [hunk 1]
```diff
@@ -6,6 +6,10 @@
  * @param height
  * @constructor
  */
+
+var _ = require('underscore');
+var jQuery = require('jquery');
```
Comment (dariocravero) [inline src/sdk/helpers/rect.js:11]: `jQuery` is generally required as `$`. There's nothing else using it there so it should be fine I guess if it stays as it is.

File: src/sdk/helpers/smil-document-parser.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
+//  Copyright (c) 2014 Readium Foundation and/or its licensees. All rights reserved.
```
Comment (dariocravero) [inline src/sdk/helpers/smil-document-parser.js]: We'll have to normailse the spaces here. [js-beautify](https://www.npmjs.com/package/js-beautify) comes in handy for that.

File: src/sdk/helpers/smil-document-parser.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+//  1. Redistributions of source code must retain the above copyright notice, this 
+//  list of conditions and the following disclaimer.
+//  2. Redistributions in binary form must reproduce the above copyright notice, 
+//  this list of conditions and the following disclaimer in the documentation and/or 
+//  other materials provided with the distribution.
+//  3. Neither the name of the organization nor the names of its contributors may be 
+//  used to endorse or promote products derived from this software without specific 
+//  prior written permission.
+
+define(['require', 'module', 'jquery', 'underscore'], function (require, module, $, _) {
```
Comment (dariocravero) [inline src/sdk/helpers/smil-document-parser.js]: We need to replace `define` for the requires.
Something like this should do:

```
var _ = require('underscore');
var $ = require('jquery);
```

File: src/sdk/helpers/smil-document-parser.js [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                if (arr.length > 0) {
+                    hours = parseFloat(arr.pop());
+                }
+            }
+        }
+        var total = hours * 3600 + mins * 60 + secs;
+        return total;
+    }
+    
+    return SmilDocumentParser;
```
Comment (dariocravero) [inline src/sdk/helpers/smil-document-parser.js]: We need to export `SmilDocumentParser` like `module.exports = SmilDocumentParser`. The wrapping function of `define` isn't needed.

File: src/sdk/models/smil-model.js [hunk 5]
```diff
@@ -0,0 +1,760 @@
+//  LauncherOSX
```
Comment (dariocravero) [inline src/sdk/models/smil-model.js]: We don't need this file. This is all done [here]https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/index.js]. Just `require('sdk/models/smil');`.

**Comment by dariocravero** [comment]
Hey @paulinthought would you have a look at those comments? You may also want to clean up the commit and either chunk it or put it all in one, [here's](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit) a good explanation on how to go about it. Thanks :)

File: src/sdk/models/smil-model.js (hunk 5 above)
Comment (paulinthought) [inline src/sdk/models/smil-model.js]: I think SmilModel is needed. The index.js creates a new instance of SmilModel which is extended in the index file but originates in the smil-model.js

Comment (dariocravero) [inline src/sdk/models/smil-model.js]: [And it is there](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/smil-node.js). The public interface is provided by `Smil` and not by the node, the node is required by more complex nodes such as [MediaNode](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/media-node.js#L1) which in turn construct both [AudioNode](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/audio-node.js#L1) and [TextNode](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/text-node.js#L1) which are required by [Smil](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/index.js#L29-L30).

**Comment by dariocravero** [comment]
So two files need to go away: `src/sdk/helpers/smil-document-parser.js` and `src/sdk/models/smil-model.js`. The first one lives at `https://github.com/dariocravero/readium-js/blob/master/src/epub/smil-document-parser.js` and the second one lives at `https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/index.js#L32`.

It seems that I renamed `SmilModel` to `Smil`, my bad.
You'd have to rename it like that, i.e., to have the folder [src/sdk/models/smil](https://github.com/dariocravero/readium-js/tree/master/src/sdk/models/smil) renamed to `smil-model` and the function renamed from `Smil` to `SmilModel` [here](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/index.js#L32) and [here](https://github.com/dariocravero/readium-js/blob/master/src/sdk/models/smil/index.js#L310).

You'd also have to require it at `src/sdk/models/media-overlay.js`.

**Pull request closed without merging**